byron/ledger/executable-spec/src/Ledger/Update/Generators.hs

```diff
@@ -3,33 +3,43 @@
 
 -- | Generators for the 'Ledger.Update' values.
 module Ledger.Update.Generators
-  ( pparamsGen
-  , protVerGen
-  -- PVBUMP environment generators
-  , pvbumpAfter2kEnvGen
-  , pvbumpEmptyListEnvGen
-  , pvbumpBeginningsEnvGen
-  -- PVBUMP state generators
-  , pvbumpStateGen
+  ( pparams
+  , protVer
+  -- PVBUMP judgement context generators
+  , emptyPVUpdateJC
+  , beginningsNoUpdateJC
+  , lastProposalJC
+  -- UPIState-related generators
+  , apName
+  , apVer
+  , metadata
+  -- UPIEC judgement context generators
+  , noProtVerChangeJC
+  , protVerChangeJC
   )
 where
 
-import           Control.State.Transition (Environment, State)
+import           Control.State.Transition (Environment, State, Signal)
+import           Data.Map.Strict (Map)
+import           Data.Set (Set)
 import           Data.Word (Word64)
 import           Hedgehog
 import           Hedgehog.Gen.Aux (doubleInc)
 import qualified Hedgehog.Gen    as Gen
 import qualified Hedgehog.Range  as Range
-import           Ledger.Core (Slot(..), SlotCount(..), BlockCount(..))
-import           Ledger.Core.Generators (slotGen)
+import           Ledger.Core (Slot(..), SlotCount(..), BlockCount(..), VKeyGenesis(..), minusSlot)
+import qualified Ledger.Core.Generators as CG
 import           Ledger.GlobalParams (k)
-import           Ledger.Update (ProtVer(..), PParams(..), PVBUMP)
+import           Ledger.Update (ProtVer(..), PParams(..), PVBUMP, ApName(..), ApVer(..), Metadata(..), UpId(..), UPIState, UPIEC)
 import           Numeric.Natural (Natural)
 
 
+-- | A judgement context of an STS
+type JC sts = (Environment sts, State sts, Signal sts)
```

Comment:
I'd rather be explicit, and use the rhs instead of the type alias. But if we go down this road, we might want to put Jesus Christ in the `small-steps` package.

Reply:
There are quite a few STSes and their judgement contexts are a regular pattern appearing in the code, hence it makes perfect sense to abbreviate this.